### DISCUSSION TITLE
G6 and Xiaomi12 data collector and BG readings reliability issue

### DISCUSSION THREAD

**Discussion by karzyg555** [NightscoutFoundation/xDrip#3574] [Bug reports]
Hi,

I'm new Dexcom G6 user, so far I've used Libre2 sensors for the past few years as it allows WearOS data collector. 
I've decided to try G6 sensors to compare both sensors BG readings precision's on my own. I use Xiaomi12 phone and latest XDrip version (system status attached below) and I've read your documentation how to configure X-Drip to receive Dexcom BG readings. 

The issue I experience right now is missing reading from G6 sensor (gap in BG data). I can see constant readings every 5 min e.g. for 1 hour and then it suddenly stops and there is no new BG readings for following 1 hour or so. To restore BG readings retrieval I've to restart collector or open XDrip. It sometimes helps, but sometimes it doesn't and I've to wait another X minutes when it "surprisingly" starts to work again. 

It looks to me like XDrip app goes into inactivity/frozen mode on phone, when I don't use phone. I've checked battery optimization settings as well other Android settings and nothing indicates, that app is being killed or moved into "sleep" mode. As example in the morning after I woke up I see no BG readings since more than hundreds of minutes. 
I believe issue is with my app settings, but I've verified all settings I could think of. Your help will appreciated :) Below you will find screenshot with my XDrip settings.
![xdrip1](https://github.com/user-attachments/assets/cbb95a71-442e-4068-889c-8850b1184570)
![xdrip2](https://github.com/user-attachments/assets/61c7d021-3596-494c-b9a1-3775a48d729e)
![xdrip3](https://github.com/user-attachments/assets/f190d136-ed89-406e-933b-04223d3b44d5)
![xdrip4](https://github.com/user-attachments/assets/217f1efa-067f-4548-a9d5-afc9f04cbe94)
![xdrip5](https://github.com/user-attachments/assets/6f8f357e-7edb-43bb-96e9-58b1d87f40f6)
![xdrip6](https://github.com/user-attachments/assets/bb66a1a1-4971-46b6-a09b-2beed3ff08d9)
![xdrip7](https://github.com/user-attachments/assets/f5693934-1b23-4acc-92cc-1ba5d81e268d)
![xdrip8](https://github.com/user-attachments/assets/ac5501d9-1e2a-42ad-a23e-9b9b0612ff4a)

**Comment by Navid200** [comment]
Before we can do anything, you need to correct the settings.
First of all, all of that has been explained here in detail:  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

Looking at the settings that you have posted, the following are the ones you have incorrectly set:
You need to disable allow OB1 unbonding.
You need to enable minimize scanning.  

One mistake many Xiaomi users make is to fail to disable their phone from disabling xDrip since it runs in the background.  And your status page showing slowest wake 4 minutes is a confirmation of that.
To address that, look at this in detail and see if you can find corresponding settings on your phone:
https://dontkillmyapp.com/xiaomi  

All of the above are steps you should take first.  After you do that, it is still possible that you may have problems.  We can talk about those when you get there.

**Comment by karzyg555** [comment]
Thanks i've amended settings and it start to work, I've updated following settings:
- disable allow OB1 unbonding.
- enable minimize scanning.
- enable wake workarounds in Bluetooth settings
- enable Settings −> Less common settings −> Aggressive service restarts

Many thanks for a help it works right now reliable. 
One more question if i would like to enable wear os data collector, should i update any of above settings ? I use TicWatch Pro 3 ultra smartwatch with wearos3 and when i tried initially and i didn't work, just one value and then no refresh

**Reply by Navid200** [reply]
I was typing a message and all of a sudden everything vanished.  Did you do something?

**Reply by karzyg555** [reply]
it was not me, maybe crowdstrike issue ;)

**Reply by Navid200** [reply]
WearOS cannot reliably collect from Dexcom as far as I know.  
I don't have experience with WearOS.

**Reply by karzyg555** [reply]
i see, do you know where should i ask for wear os help ?

**Reply by Navid200** [reply]
@PhilTU

**Reply by PhilTU** [reply]
I can only suggest trying the wearos collector by turning off phone Bluetooth and seeing what happens. Watch should take over. Or set to force WearOS collection on phone.

**Comment by karzyg555** [comment]
I did it already and I got just first reading only and then nothing happens. 
From my observation I found out, that I have to remove Dexcom transmitter bluetooth device from remembered devices on the watch and then watch asks me again to trust Dexcom transmitter and I got just one new BG reading.
I wonder if there are any other settings outside the one you mentioned to try reconfigure wearos collector.

Kz
